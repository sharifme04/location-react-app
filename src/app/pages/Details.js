import React from "react";
import { useSelector } from "react-redux";
import {
  Header,
  HeaderLayout,
  Breadcrumb,
  DistanceBar,
  DetailsCard,
} from "../components";
import carIcon from "../assets/icons/car-icon.png";
import logoIcon from "../assets/icons/logo-icon.png";

const Details = () => {
  const { tripSearch, isLoading } = useSelector(
    (state) => state.tripSearchReducer || {}
  );

  console.log("tripSearch", tripSearch);

  return isLoading ? (
    <div className="d-flex justify-content-center custom-center">
      <div className="spinner-border custom-border" role="status">
        <span className="visually-hidden">Loading...</span>
      </div>
    </div>
  ) : (
    <div>
      <HeaderLayout>
        <Breadcrumb />
        <Header
          title={"Umweltrechner"}
          subt1={"Penden"}
          subt2={"210 Tag"}
          subt3={"mit"}
          subt4={"Miteklase"}
          subt5={"von"}
          subt6={"Tam"}
          subt7={"bis"}
          subt8={"Statmit"}
        />
      </HeaderLayout>
      <DistanceBar />
      <div className="card-container">
        <div className="details-card-subtitle">
          <div className="custom-card-subtitle">
            Auswirkungen auf die Umwelt
          </div>
          <div className="custom-card-group">
            <DetailsCard
              logo={"logo"}
              name={"Auto"}
              weight={"2.032 kg"}
              time={"CO2 / Jahr"}
              customClass={"details-card-custom-style1"}
              icon={carIcon}
            />
            <DetailsCard
              logo={"logo"}
              name={"VVS Abo"}
              weight={"758 kg"}
              time={"CO2 / Jahr"}
              bgColor={"#fc631e"}
              customClass={"details-card-custom-style2"}
              icon={logoIcon}
            />
          </div>
        </div>
        <div className="details-card-subtitle">
          <div className="custom-card-subtitle">
            Kosteneinschätzung für ein Jahr
          </div>
          <div className="custom-card-group">
            <DetailsCard customClass={"details-card-custom-style1"} />
            <DetailsCard customClass={"details-card-custom-style2"} />
          </div>
        </div>
      </div>
      <div className="next-step">
        <p>Wie entstehen diese Werte?</p>
      </div>
      <div className="card-layers">
        <div>
          <div className="lower-square"></div>
          <div className="upper-square"></div>
        </div>
      </div>
    </div>
  );
};

export default Details;
