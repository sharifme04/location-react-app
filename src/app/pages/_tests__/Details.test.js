import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { MemoryRouter } from "react-router-dom";
import Details from "../Details";

const mockStore = configureMockStore();
const store = mockStore({});

test("snapshot form renders", () => {
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Details />
      </MemoryRouter>
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});