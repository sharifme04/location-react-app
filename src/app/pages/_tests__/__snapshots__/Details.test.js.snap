// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot form renders 1`] = `
<div>
  <div
    className="headerContainer"
  >
    <div>
      <nav
        aria-label="breadcrumb"
      >
        <ol
          className="breadcrumb"
        >
          <li
            className="breadcrumb-item"
          >
            Rundum mobil
          </li>
          <li
            className="breadcrumb-item"
          >
            Preis- und Umweltvergleich
          </li>
        </ol>
      </nav>
    </div>
    <div>
      <div>
        <span
          className="title"
        >
          Umweltrechner
        </span>
      </div>
      <div
        className="subtitle"
      >
        <span>
          Penden
        </span>
        <span>
          <a
            className="link-subtitle"
            href="#"
          >
            210 Tag
          </a>
           
        </span>
        <span>
          mit
        </span>
        <span>
          <a
            className="link-subtitle"
            href="#"
          >
            Miteklase
          </a>
           
        </span>
        <span>
          von
        </span>
        <span>
          <a
            className="link-subtitle"
            href="#"
          >
            Tam
          </a>
           
        </span>
        <span>
          bis
        </span>
        <span>
          <a
            className="link-subtitle"
            href="#"
          >
            Statmit
          </a>
           
        </span>
      </div>
    </div>
  </div>
  <div
    className="bar-container"
  >
    <div
      className="distance-bar"
    >
      <div
        className="car"
      >
        <svg>
          car-icon.svg
        </svg>
        <p>
          24.1 km
        </p>
      </div>
      <div
        className="bus"
      >
        <svg>
          bus-icon.svg
        </svg>
        <p>
          24.8 km
        </p>
      </div>
    </div>
    <div
      className="triangle-container"
    >
      <div
        className="triangle"
      />
    </div>
  </div>
  <div
    className="card-container"
  >
    <div
      className="details-card-subtitle"
    >
      <div
        className="custom-card-subtitle"
      >
        Auswirkungen auf die Umwelt
      </div>
      <div
        className="custom-card-group"
      >
        <div
          className="details-card details-card-custom-style1"
          style={
            Object {
              "backgroundColor": "",
            }
          }
        >
          <div
            className="details-card-title"
          >
            <img
              alt="logo-icon"
              height="24"
              src="car-icon.png"
              width="24"
            />
            <div
              className="card-inner-title"
            >
               
              Auto
            </div>
          </div>
          <div
            className="card-weight"
          >
            2.032 kg
          </div>
          <div
            className="card-time"
          >
            CO2 / Jahr
          </div>
        </div>
        <div
          className="details-card details-card-custom-style2"
          style={
            Object {
              "backgroundColor": "#fc631e",
            }
          }
        >
          <div
            className="details-card-title"
          >
            <img
              alt="logo-icon"
              height="24"
              src="logo-icon.png"
              width="24"
            />
            <div
              className="card-inner-title"
            >
               
              VVS Abo
            </div>
          </div>
          <div
            className="card-weight"
          >
            758 kg
          </div>
          <div
            className="card-time"
          >
            CO2 / Jahr
          </div>
        </div>
      </div>
    </div>
    <div
      className="details-card-subtitle"
    >
      <div
        className="custom-card-subtitle"
      >
        Kosteneinschätzung für ein Jahr
      </div>
      <div
        className="custom-card-group"
      >
        <div
          className="details-card details-card-custom-style1"
          style={
            Object {
              "backgroundColor": "",
            }
          }
        >
          <div
            className="details-card-title"
          />
        </div>
        <div
          className="details-card details-card-custom-style2"
          style={
            Object {
              "backgroundColor": "",
            }
          }
        >
          <div
            className="details-card-title"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="next-step"
  >
    <p>
      Wie entstehen diese Werte?
    </p>
  </div>
  <div
    className="card-layers"
  >
    <div>
      <div
        className="lower-square"
      />
      <div
        className="upper-square"
      />
    </div>
  </div>
</div>
`;
